Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -354,13 +524,53 @@ bool isFunctionJson(string fullName)
             }
         }
 
-        private async Task<string> DeployZipLocally(ZipDeploymentInfo zipDeploymentInfo, ITracer tracer)
+        private async Task ArtifactFetch(IRepository repository, DeploymentInfoBase deploymentInfo, string targetBranch, ILogger logger, ITracer tracer)
+        {
+            var artifactDeploymentInfo = (ArtifactDeploymentInfo)deploymentInfo;
+            //Fetches content from URL if ArtifactURL exists
+            HttpContent content = !string.IsNullOrEmpty(artifactDeploymentInfo.ArtifactURL)
+                ? await DeploymentHelper.GetArtifactContentFromURL(artifactDeploymentInfo, tracer)
+                : Request.Content;
+
+            //repository.RepositoryPath is the directory the deployment pipeline expects the files that will be deployed to be
+            var tempDirPath = repository.RepositoryPath;
+            if (deploymentInfo.CleanupTargetDirectory)
+            {
+                var targetInfo = FileSystemHelpers.DirectoryInfoFromDirectoryName(tempDirPath);
+                if (targetInfo.Exists)
+                {
+                    // If tempDirPath already exists, rename it so we can delete it concurrently with
```

Comment:
This comment says that you want to start DeleteFilesAndDirsExcept asap, but that is not what you are doing.